File: SWEA/D2/1961. 숫자 배열 회전/숫자 배열 회전.py
def rotate(arr):
    new_arr = [[0]*N for _ in range(N)]
    for i in range(N):
        for j in range(N-1, -1, -1):
            new_arr[i][N-j-1] = arr[j][i]
    return new_arr

def printForm(res):
    new_res = []
    for i in range(N):
        temp = ''
        for j in range(N):
            temp += str(res[i][j])
        new_res.append(temp)
    return new_res

T = int(input())
for tc in range(1, T + 1):
    N = int(input())
    arr = [list(map(int, input().split())) for _ in range(N)]
    arr90 = rotate(arr)
    arr180 = rotate(arr90)
    arr270 = rotate(arr180)
    res90 = printForm(arr90)
    res180 = printForm(arr180)
    res270 = printForm(arr270)
    print(f'#{tc}')
    for i in range(N):
        print(res90[i], res180[i], res270[i])

File: SWEA/D2/1961. 숫자 배열 회전/숫자 배열 회전.java
import java.util.Scanner;
import java.io.FileInputStream;

class Solution
{
    // 회전 함수
    public static int[ ][ ] rotate(int[ ][ ] arr) throws Exception
    {
        int size = arr[0].length;
        int[ ][ ] rotate_arr = new int[size][size];
        for(int row = 0; row < size; row++)
        {
            for(int col = 0; col < size; col++)
            {
                rotate_arr[row][size-1-col] = arr[col][row];
            }
        }
        return rotate_arr;
    }
    
	public static void main(String args[]) throws Exception
	{
		Scanner sc = new Scanner(System.in);
		int T;
		T=sc.nextInt();
        // 테스트 케이스 1개씩
		for(int test_case = 1; test_case <= T; test_case++)
		{
            // N 입력받음
            int N = sc.nextInt();
            // 배열 선언 및 초기화
            int[ ][ ] arr = new int[N][N];
            for(int i = 0; i < N; i++) 
            {
                for(int j = 0; j < N; j++) 
                {
                    arr[i][j] = sc.nextInt();
                }
            }
            // 회전 3번
            int[ ][ ] arr_90 = rotate(arr);
            int[ ][ ] arr_180 = rotate(arr_90);
            int[ ][ ] arr_270 = rotate(arr_180);
            // 출력
            System.out.print("#" + test_case + "\n");
            for(int i = 0; i < N; i++) 
            {
                for(int j = 0; j < N; j++) 
                {
                    System.out.print(arr_90[i][j]);
                }
                System.out.print(" ");
                for(int j = 0; j < N; j++) 
                {
                    System.out.print(arr_180[i][j]);
                }
                System.out.print(" ");
                for(int j = 0; j < N; j++) 
                {
                    System.out.print(arr_270[i][j]);
                }                
                System.out.print("\n");
            }
		}
	}
}